.github/workflows/pr-quality-check.yml

```
@@ -0,0 +1,131 @@
+
+          # Warn if PR is too large
+          if [ $TOTAL_CHANGES -gt 1000 ]; then
+            echo "⚠️ **Warning:** This PR is quite large ($TOTAL_CHANGES lines changed). Consider breaking it into smaller PRs for easier review." >> $GITHUB_STEP_SUMMARY
+          elif [ $TOTAL_CHANGES -gt 500 ]; then
+            echo "⚠️ **Note:** This PR is moderately sized ($TOTAL_CHANGES lines changed)." >> $GITHUB_STEP_SUMMARY
+          else
+            echo "✅ **Good:** This PR is reasonably sized ($TOTAL_CHANGES lines changed)." >> $GITHUB_STEP_SUMMARY
+          fi
+
+      - name: Check PR Title Format
+        run: |
```

actionlint (1.7.9)

"github.event.pull_request.title" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)